src/app/views/settings/Settings.tsx

```diff
@@ -0,0 +1,164 @@
+import {
+  ChoiceGroup, Dialog, DialogType, IconButton
+} from 'office-ui-fabric-react';
+import React, { Component } from 'react';
+import { injectIntl } from 'react-intl';
+import { connect } from 'react-redux';
+import { bindActionCreators, Dispatch } from 'redux';
+import * as authActionCreators from '../../services/actions/auth-action-creators';
+import * as themeAtionCreators from '../../services/actions/theme-action-creator';
+
+export interface ISettingsProps {
+  actions?: {
+    signOut: Function;
+    changeTheme: Function;
+  };
+  intl?: {
+    message: object;
+  };
+  authenticated: boolean;
+  appTheme: string;
+}
+
+class Settings extends Component<ISettingsProps, any> {
+  constructor(props: ISettingsProps) {
+    super(props);
+    this.state = {
+      hideThemeChooserDialog: true,
+    };
+  }
+
+  public toggleThemeChooserDialogState = () => {
+    this.setState({ hideThemeChooserDialog: !this.state.hideThemeChooserDialog });
+  }
+
+  public handleSignOut = () => {
+    this.props.actions!.signOut();
+  }
+
+  public handleChangeTheme = (event: any, option: any) => {
+    const newTheme = option.key;
```

Comment:
I don't think it's a good idea to reload the application when the theme changes. We don't have to reload TryIt for it to pick a new theme. You might want to call the `loadGETheme` function here and fix the components that are not picking up the theme.

Reply:
On this and this:

> Let's not reload the page on theme change. We might find the loagGETheme function useful.

We actually do need to reload. The reason it works for try it is because once the theme changes on the dev portal it loads a new version of GE with the theme parameters.

loadGETheme does not re-render the screen

Reply:
Nope, it doesn't reload GE, refer to `App.tsx`.